quinn-udp/src/lib.rs

```diff
@@ -38,9 +38,20 @@ use std::{
 };
 
 #[cfg(all(feature = "direct-log", not(feature = "tracing")))]
-use log::warn;
+use log;
 #[cfg(feature = "tracing")]
-use tracing::warn;
+use tracing as log;
```

Comment:
This punning makes it easy to inadvertently use an item that doesn't exist under a different feature configuration. A more robust approach would be to have an unconditionally defined internal log module that only exports exactly the intended interface, implemented as macros that are unconditionally defined, and only have variation in their bodies.